code/modules/VR/vr_avatar.dm

```diff
@@ -0,0 +1,87 @@
+
+
+/mob/living/carbon/human/virtual_reality
+	var/mob/living/carbon/human/real_me //The human controlling us, can be any human (including virtual ones... inception...)
+	var/datum/action/quit_vr/quit_action
+	var/datum/action/back_to_lobby/back_to_lobby
+	var/datum/action/detach_from_avatar/detach_from_avatar
+
+/mob/living/carbon/human/virtual_reality/New()
+	quit_action = new()
+	quit_action.Grant(src)
+	back_to_lobby = new()
+	back_to_lobby.Grant(src)
+	detach_from_avatar = new()
+	detach_from_avatar.Grant(src)
+	..()
+
+/mob/living/carbon/human/virtual_reality/death()
+	return_to_lobby()
+	..()
```

Comment:
Yeah, this is a good idea due to me wanting to make death return 1 if the person actually died from that call, while returning 0 if the person did not die, either due to invulnerability or already being dead.